Assets/Scripts/Mixed/Systems/KinematicCharacterControllerSystems.cs

```diff
@@ -3,23 +3,31 @@
 using Unity.Entities;
 using Unity.Jobs;
 using Unity.Mathematics;
+using Unity.NetCode;
 using Unity.Physics;
 using Unity.Physics.Systems;
 using Unity.Transforms;
 
 namespace PropHunt.Mixed.Systems
 {
+    /// <summary>
+    /// Update groups for things to compute before physics step computation
+    /// </summary>
+    [UpdateInGroup(typeof(FixedStepSimulationSystemGroup))]
+    [UpdateBefore(typeof(BuildPhysicsWorld))]
+    public class KCCPreUpdateGroup : ComponentSystemGroup { }
+
     /// <summary>
     /// System group for all Kinematic Character Controller Actions
     /// </summary>
-    [UpdateAfter(typeof(FixedStepSimulationSystemGroup))]
-    [UpdateBefore(typeof(PushForceGroup))]
+    [UpdateInGroup(typeof(FixedStepSimulationSystemGroup))]
+    [UpdateAfter(typeof(EndFramePhysicsSystem))]
     public class KCCUpdateGroup : ComponentSystemGroup { }
 
     /// <summary>
     /// Updates the grounded data on a kinematic character controller
     /// </summary>
-    [UpdateInGroup(typeof(KCCUpdateGroup))]
+    [UpdateInGroup(typeof(KCCPreUpdateGroup))]
```

Comment:
This is the big change to fix updating around the physics system.

I found this solution by investigating the KCC using the created test objects.